ingesters/s3Ingester/manager.go

```diff
@@ -19,28 +19,37 @@ import (
 const (
 	manualTickerInterval = time.Minute
 	ERROR_BACKOFF        = 5 * time.Second
+	QUEUE_DEPTH          = 1000
 )
 
 var (
 	errEmptyBucket = errors.New("empty bucket name")
 	errEmptyKey    = errors.New("empty key name")
 )
 
-func start(wg *sync.WaitGroup, ctx context.Context, buckets []*BucketReader, sqsS3 []*SQSS3Listener, ot *objectTracker, lg *log.Logger) (err error) {
+func start(wg *sync.WaitGroup, ctx context.Context, buckets []*BucketReader, sqsS3 []*SQSS3Listener, ot *objectTracker, lg *log.Logger, numWorkers int) (err error) {
 	if len(buckets) != 0 {
 		wg.Add(1)
-		go manualScanner(wg, ctx, buckets, ot, lg)
+		go manualScanner(wg, ctx, buckets, ot, lg, numWorkers)
 	}
 	for _, v := range sqsS3 {
 		wg.Add(1)
-		go sqsS3Routine(v, wg, ctx, lg)
+		go sqsS3Routine(v, wg, ctx, lg, numWorkers)
 	}
 	return
 }
 
-func sqsS3Routine(s *SQSS3Listener, wg *sync.WaitGroup, ctx context.Context, lg *log.Logger) {
+func sqsS3Routine(s *SQSS3Listener, wg *sync.WaitGroup, ctx context.Context, lg *log.Logger, numWorkers int) {
 	defer wg.Done()
 
+	// create workers
+	var workerWg sync.WaitGroup
+	queue := make(chan *sqs.Message, QUEUE_DEPTH)
```

Comment:
I don't see that this ever actually gets closed?  how are workers shutdown on this code path?

Reply:
good catch -- PTAL